lib/utils/addEntries.js

```diff
@@ -159,6 +152,57 @@ function addEntries(config, options, server) {
         config.plugins.push(new webpack.HotModuleReplacementPlugin());
       }
     }
+
+    // if there are no additional entries to be added, nothing
+    // below is needed
+    if (!additionalEntries.length) {
+      return;
+    }
+
+    if (webpack.version[0] === '5') {
+      // EntryPlugin is only available for webpack@5
+      // @ts-ignore
+      const EntryPlugin = require('webpack/lib/EntryPlugin'); // eslint-disable-line import/no-unresolved
+      // for webpack@5, we use a plugin to add global entries
+      compiler.hooks.make.tapAsync(
+        {
+          name: 'webpack-dev-server',
+          stage: -100,
```

Comment:
@evilebottnawi @sokra I've added the tapAsync method here, but the issue I've found now is that something like this where multiple Servers are attached to a compiler no longer works:

```
const compiler = webpack(config);
const server1 = new Server(compiler, options1);
const server2 = new Server(compiler, options2);
```

as it results in something like `Error: Conflicting entry option static = [object Object] vs [object Object]`

This worked before, as there were tests that did this which are now broken. (These tests would usually close the old server before a new one was attached)

Should we go about this by making multiple servers on one compiler discouraged, or maybe a hacky method of disabling the old hook taps like in https://github.com/webpack/tapable/issues/109 after the server is closed?

Reply:
Or I could also try to prevent duplicate entries by looking at the compilation globalEntry object.

Reply:
I added the hacky method for disabling old hook taps, but haven't added tests for it yet

Reply:
@Loonride I think we should prevent duplicate, in theory we can have property on `compiler` to achieve this, we use same logic here https://github.com/webpack/webpack-dev-middleware/blob/master/src/utils/setupWriteToDisk.js#L9 (but it only for webpack@4)

Reply:
I think it is more clean hack :smile:

Reply:
@evilebottnawi I agree it is more clean, but one issue I now see is that if the user does:

```
const compiler = webpack(config);
const server1 = new Server(compiler, options1);
server1.close(() => {
  const server2 = new Server(compiler, options2);
});
```

If the `options2` object results in a new set of entries, that will not be reflected in the compilation, since the old make callback would still be used.